File: java/src/main/java/de/gdata/vaas/messages/VaasVerdict.java
package de.gdata.vaas.messages;

import lombok.Getter;
import lombok.NonNull;

@Getter
public class VaasVerdict {
    @NonNull
    String sha256;
    @NonNull
    Verdict verdict;
    String detection;
    String fileType;
    String mimeType;

    public VaasVerdict(UrlReport urlReport) {
        this.sha256 = urlReport.sha256;
        this.verdict = urlReport.verdict;
        this.detection = urlReport.detection;
        this.fileType = urlReport.fileType;
        this.mimeType = urlReport.mimeType;
    }

    public VaasVerdict(FileReport fileReport) {
        this.sha256 = fileReport.sha256;
        this.verdict = fileReport.verdict;
        this.detection = fileReport.detection;
        this.fileType = fileReport.fileType;
        this.mimeType = fileReport.mimeType;
    }

    public static VaasVerdict From(UrlReport urlReport) {
        return new VaasVerdict(urlReport);
    }

    public static VaasVerdict From(FileReport fileReport) {
        return new VaasVerdict(fileReport);
    }
}

File: dotnet/Vaas/src/Vaas/VaasVerdict.cs
using System.Text.Json;
using Vaas.Messages;

namespace Vaas;

public class VaasVerdict
{
    public required string Sha256 { get; init; }
    public Verdict Verdict { get; init; }
    public string? Detection { get; init; }
    public string? MimeType { get; init; }
    public string? FileType { get; init; }

    public static VaasVerdict From(FileReport fileReport)
    {
        return new VaasVerdict
        {
            Sha256 = fileReport.Sha256,
            Verdict = fileReport.Verdict,
            Detection = fileReport.Detection,
            MimeType = fileReport.MimeType,
            FileType = fileReport.FileType,
        };
    }

    public static VaasVerdict From(UrlReport urlReport)
    {
        return new VaasVerdict
        {
            Sha256 = urlReport.Sha256,
            Verdict = urlReport.Verdict,
            Detection = urlReport.Detection,
            MimeType = urlReport.MimeType,
            FileType = urlReport.FileType,
        };
    }

    public override string ToString() => JsonSerializer.Serialize(this);
}
